cmd/metricscollector/v1beta1/file-metricscollector/main.go

```diff
@@ -107,7 +107,7 @@ var (
 	metricFilters        = flag.String("f", "", "Metric filters")
 	pollInterval         = flag.Duration("p", common.DefaultPollInterval, "Poll interval between running processes check")
 	timeout              = flag.Duration("timeout", common.DefaultTimeout, "Timeout before invoke error during running processes check")
-	waitAll              = flag.Bool("w", common.DefaultWaitAll, "Whether wait for all other main process of container exiting")
+	waitAllProcesses     = flag.String("w", common.DefaultWaitAllProcesses, "Whether wait for all other main process of container exiting")
```

Comment:
Do we need to keep this flag `String`, or we can use `Bool`?
I assume this flag can be only `true` or `false`.

Reply:
I had problems with parsing booleans when a user has not specified the field in the katib-config. Do you know of a good way to do it?
I could leave it string in the config and parse it it in `GetMetricsCollectorConfigData`, would that be better?

Reply:
@robbertvdg It seems that empty `bool` in Go is `false`. So when user doesn't specify any values in config we send "false" to the metrics collector.
Maybe we can define `MetricsCollectorConfig` like this:
```go
type MetricsCollectorConfig struct {
	Image            string                      `json:"image"`
	ImagePullPolicy  corev1.PullPolicy           `json:"imagePullPolicy"`
	Resource         corev1.ResourceRequirements `json:"resources"`
	WaitAllProcesses *bool                      `json:"waitAllProcesses"`
}
```
In that case, you can set `-w` flag if `WaitAllProcesses != nil`. Otherwise the default value is set.

In Katib config I was able to define bool value like this:
```json
 "StdOut": {
        "image": "docker.io/andreyvelichkevich/file-metrics-collector",
        "imagePullPolicy": "Always",
        "waitAllProcesses": false
      },
```

Reply:
The problem lies with the `v1.Container.args` which is `string[]`. Therefore every container argument has to be a string, which I think is not compatible with the flag.Bool() argument (I tested it, doesn't work).

Reply:
@robbertvdg Got it.
In that case can we keep `WaitAllProcesses *bool` in `MetricsCollectorConfig`, convert it to string here: 
https://github.com/kubeflow/katib/blob/3f5d81302a9ff2f43c30ab70bf9c6d0bc49b525e/pkg/webhook/v1beta1/pod/inject_webhook.go#L299
Then, use the String type like you did in [Metrics Collector](https://github.com/kubeflow/katib/blob/3f5d81302a9ff2f43c30ab70bf9c6d0bc49b525e/cmd/metricscollector/v1beta1/file-metricscollector/main.go#L110) and convert it to Bool after.

In that case, we can avoid situation when user specify wrong values in Katib config.

What do you think @robbertvdg @gaocegege ?

Reply:
@andreyvelich those were my thoughts as well, will update PR soon.